### ISSUE TITLE
`mypy` silent about `Never`, `NoReturn`

### ISSUE THREAD

**Issue by alexei** [python/mypy#16389]
**Bug Report**

Basically the title.

**To Reproduce**

```python
from typing import Never, NoReturn


def foo() -> NoReturn:
    raise NotImplementedError()


def bar() -> Never:
    raise NotImplementedError()


foo()
bar()
```

See https://mypy-play.net/?mypy=1.6.1&python=3.11&gist=21156fba6248fbb3e0a604933dde9e8d

**Expected Behavior**

I would expect mypy to report when e.g. calling a function with `Never` or `NoReturn` return type.

**Actual Behavior**

```
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 1.6.1
- Python version used: 3.11

**Comment by JelleZijlstra** [comment]
Calling a function with a `NoReturn` return type is not an error. The canonical example is `sys.exit`.

You may be interested in the `--warn-unreachable` flag, which makes mypy flag unreachable code (including code after a call to a NoReturn function).

**Comment by alexei** [comment]
What about `Never`?

> This can be used to define a function that should never be called

Still behind [flag](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-warn-unreachable)? Docs seem to only mention `NoReturn`

**Comment by JelleZijlstra** [comment]
Never and NoReturn are exact synonyms.

**Comment by alexei** [comment]
Thanks @JelleZijlstra
